src/main/java/com/sintern/service/ApplicationServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.sintern.service;
+
+import com.sintern.domain.entity.Application;
+import com.sintern.domain.entity.FileEntity;
+import com.sintern.domain.entity.OpenInternPosition;
+import com.sintern.domain.entity.Student;
+import com.sintern.exception.CVNotFoundException;
+import com.sintern.exception.EntityNotFoundException;
+import com.sintern.repository.ApplicationRepository;
+import com.sintern.repository.FileRepository;
+import com.sintern.repository.OpenInternPositionRepository;
+import com.sintern.repository.StudentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class ApplicationServiceImpl implements ApplicationService {
+
+    private final ApplicationRepository applicationRepository;
+    private final StudentRepository studentRepository;
+    private final OpenInternPositionRepository openInternPositionRepository;
+    private final FileRepository fileRepository;
+
+    @Autowired
+    public ApplicationServiceImpl(ApplicationRepository applicationRepository,
+                                  StudentRepository studentRepository,
+                                  OpenInternPositionRepository openInternPositionRepository,
+                                  FileRepository fileRepository) {
+        this.applicationRepository = applicationRepository;
+        this.studentRepository = studentRepository;
+        this.openInternPositionRepository = openInternPositionRepository;
+        this.fileRepository = fileRepository;
+    }
+
+    @Override
+    public void addApplication(UUID studentID, UUID openInternPositionID, String description) {
+        Student student = studentRepository.findStudentById(studentID);
+        if(student == null)
+            throw new EntityNotFoundException("Student with given id does not exist!");
```

Comment:
**Java guidelines** recommend adding **{ }** even for a **single-line** code if statement. Please add **{ }** to the lines below as well.

Reply:
Solved!